### ISSUE TITLE
IMAP4 does not allow fetch of BODY[] (imap4.py v1.131)

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7466]
|<img alt="anadelonbrin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| anadelonbrin reported|
|-|-|
|Trac ID|trac#131|
|Type|defect|
|Created|2003-07-27 11:21:59Z|

<details><summary>Searchable metadata</summary>

```
trac-id__131 131
type__defect defect
reporter__anadelonbrin anadelonbrin
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1059304919000000 1059304919000000
changetime__1465029053688350 1465029053688350
version__ 
owner__ 
cc__exarkun cc__anadelonbrin
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="anadelonbrin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|anadelonbrin commented|
|-|-|

```
#!html
<pre>
imap4.py currently doesn't allow a fetch of BODY[] or 
BODY.PEEK[] (the state_section function requires HEADER*, 
TEXT or MIME).  BODY[] and BODY.PEEK[] are legal imap 
requests (and quite different from BODY and BODYSTRUCTURE).

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="anadelonbrin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|anadelonbrin commented|
|-|-|

```
#!html
<pre>
Note also that the handling of BODY is incorrect.  BODY is 
not the same as BODY[], it is the same as BODYSTRUCTURE, 
except that it doesn't include the ext stuff.  I have a 
working copy if you want a diff.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
BODY[] and BODY.PEEK[] are now allowed.  BODY[] is handled
as BODYSTRUCTURE, but without extension fields; BODY is now
handled as RFC822.

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="anadelonbrin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|anadelonbrin commented|
|-|-|

```
#!html
<pre>
Or rather, BODY[] is RFC822, and BODY is BODYSTRUCTURE but 
without extension fields.  Seems like the command is wrong, 
but the code is right, so this is just in case anyone else 
reads this.

Thanks for making the correction (and the others).

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_6"></a>|@cyli commented|
|-|-|

In changeset 19ddd2db1f79bd2f5f9fd9d1d4c47f8ab04fb0d8
```
#!CommitTicketReference repository="" revision="19ddd2db1f79bd2f5f9fd9d1d4c47f8ab04fb0d8"
Merge pull request #131 from rodrigc/8415-rodrigc-print: Convert from print statement to print() function in docs/words

Author: rodrigc
Reviewer: reaperhulk, cyli
Fixes: #8415

Convert from print statement to print() function in docs/words
```
